src/main/java/org/grouplens/grapht/Instantiators.java

```diff
@@ -132,21 +144,25 @@ public Class getType() {
 
     private static class ProviderInstantiator implements Instantiator {
         private final Instantiator providerInstantiator;
-
         public ProviderInstantiator(Instantiator prov) {
             providerInstantiator = prov;
         }
 
         @Override
         public Object instantiate() throws ConstructionException {
+            Logger.trace(" provider invoked "+providerInstantiator);
             Provider<?> provider = (Provider) providerInstantiator.instantiate();
-            try {
+            try{
+                // --- push providerInstantiator to MDC ----------------------------------
+                mdcContext.put(provider.getClass().toString(), provider.toString());
+                mdcContext.finish();
                 return provider.get();
-            } catch (Throwable th) {
+            }
+            catch (Throwable th) {
```

Comment:
Let's merge these lines back together - the original `} catch (Throwable th) {` is our usually style.